pkg/util/test/util.go

```diff
@@ -629,6 +632,59 @@ func FindAndCopyVsphereSecretToCustomNamespace(customNamespace string) error {
 	return nil
 }
 
+// CreateAzureCredentialEnvVarsFromSecret check if px-vsphere-secret exists and returns vSphere crendentials Env vars
+func CreateAzureCredentialEnvVarsFromSecret(namespace string) ([]v1.EnvVar, error) {
+	var envVars []v1.EnvVar
+
+	// Get PX vSphere secret
+	_, err := coreops.Instance().GetSecret(DefaultPxAzureSecretName, namespace)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			logrus.Warnf("Azure secret %s in not found in %s namespace", DefaultPxAzureSecretName, namespace)
+			return nil, nil
+		}
+		return nil, fmt.Errorf("failed to get secret %s in %s namespace, err %v", DefaultPxVsphereSecretName, namespace, err)
```

Comment:
Trying to understand the intention behind not sending error when errors.IsNotFound

Reply:
I think we should return an error. Will raise a follow up PR